src-built-in/components/appMarket/src/stores/storeActions.js

```diff
@@ -0,0 +1,288 @@
+/*!
+* Copyright 2018 by ChartIQ, Inc.
+* All rights reserved.
+*/
+import AppDirectory from '../modules/AppDirectory'
+import FDC3 from '../modules/FDC3'
+import { getStore } from './appStore';
+
+export default {
+    getApps,
+    getFilteredApps,
+    clearFilteredApps,
+    searchApps,
+    getActiveTags,
+    getTags,
+    addTag,
+    removeTag,
+    clearTags,
+    addApp,
+    removeApp,
+    fetchInstalledApps,
+    getInstalledApps
+}
+
+/**
+ * I'm thinking of using  a local distributed store and the FDC3 client
+ * Some data will be pulled from appd and some from the local store
+ * Not sure if the localstore needs to be a persistent one at some point?
+ */
+const FDC3Client = new FDC3({url: 'http://localhost:3030/v1'})
+const appd = new AppDirectory(FDC3Client);
+
+/**
+ * Private function to add an active tag. This will filter apps based on tags
+ * NOTE: This will need to use search
+ * @param {string} tag The name of the tag
+ */
+function _addActiveTag(tag) {
+    let activeTags = getStore().getValue({
+        field: 'activeTags'
+    });
+
+    let apps = getStore().getValue({
+        field: 'apps'
+    });
+
+    activeTags.push(tag);
+
+    let newApps = apps.filter((app) => {
+        for (let i = 0; i < activeTags.length; i++) {
+            let tag = activeTags[i];
+            if (app.tags.includes(tag)) {
+                return true;
+            }
+        }
+    });
+
+    getStore().setValue({
```

Comment:
Recommend using `setValues`. It allows you to pass an array of values. https://documentation.chartiq.com/finsemble/StoreModel.html#setValues